### PR TITLE
Refactor authority selection - Closes #60

### LINKED ISSUE #60: Refactor authority selection
### Description

`circuit/execution-delivery` is currently storing the keypair of validators indirectly using `pallet-im-online`, however, the current implementation isn't functional and [only selects the first authority available](https://github.com/t3rn/t3rn/pull/47/files#diff-244ca883666663d39ebffd1392faeddf6218022003d803ac69ea397ee8d3575dR551)

Expected behaviour accepts the escrow account's id (either T::AccountId or T::MultiLocation, see [connected issue](https://www.notion.so/Sprint-1-Planning-3rd-June-2021-151897d7da4f448f87c56d5f6049cf34)) and selects connected `AuthorityId`. 

```rust
// let submitter = 
	local_keys.binary_search(&escrow_account. special_convert() )
	.ok().map(|location| local_keys[location].clone())
	.ok_or("Can't match")?;

let submitter = local_keys[0].clone();
```

### Acceptance Criteria

- [ ] Select `AuthorityId` connected with an escrow account.
- [ ] The selection of `T::AuthorityId` need to be tested and mocked authorities inserted for the sake of testing.

### PR DESCRIPTION
**Pull request by beqaabu**
In this PR I: 

- Implemented a function called `select_authority` which changed the method of selection from selecting the first authority to selecting the authority with a given escrow account
- Added tests

### CODE REVIEW

File: circuit/src/execution-delivery/src/lib.rs [hunk 1]
```diff
@@ -794,6 +794,20 @@ impl<T: Config> Pallet<T> {
...
+    pub fn select_authority(escrow_account: T::AccountId) -> Result<AuthorityId, &'static str> {
+        let local_keys = AuthorityId::all();
+
+        let auth = AuthorityId::from_slice(escrow_account.encode().as_slice());
+
+        let submitter = local_keys
+            .binary_search(&auth)
+            .ok()
+            .map(|location| local_keys[location].clone())
+            .ok_or("Can't match")?;
```
Comment (MaciejBaj) [inline circuit/src/execution-delivery/src/lib.rs:806]:
```suggestion
            .ok_or("Can't match authority with for given account")?;
```

**Review by MaciejBaj — CHANGES_REQUESTED** [review]

File: circuit/src/execution-delivery/src/tests.rs [hunk 2]
```diff
@@ -1245,3 +1252,58 @@ fn it_should_throw_with_empty_io_schedule() {
         expected
     );
 }
+
+#[test]
+fn test_authority_selection() {
+    let keystore = KeyStore::new();
+
+    // Insert Alice
+    let suri_alice = "//Alice";
```
[the remark this answers is not in the document]
Comment (beqaabu) [reply circuit/src/execution-delivery/src/tests.rs]: Updated on the following commit

File: circuit/src/execution-delivery/src/lib.rs [hunk 3]
```diff
@@ -811,11 +825,10 @@ impl<T: Config> Pallet<T> {
                 .map_err(|_e| "Can't cast value in dry_run_single_contract")?,
         );
 
-        let local_keys = AuthorityId::all();
-
-        // ToDo: Select validators to submit by his public key, like:
-        // let submitter = local_keys.binary_search(&escrow_account.into()).ok().map(|location| local_keys[location].clone()).ok_or("Can't match")?;
-        let submitter = local_keys[0].clone();
+        let submitter = match Self::select_authority(escrow_account.clone()) {
```
[the remark this answers is not in the document]
Comment (beqaabu) [reply circuit/src/execution-delivery/src/lib.rs]: added `?`, removed `match` on the following commit

File: circuit/src/execution-delivery/src/tests.rs [hunk 4]
```diff
@@ -1245,3 +1252,57 @@ fn it_should_throw_with_empty_io_schedule() {
...
+    // Bob's account
+    let escrow: AccountId =
+        hex_literal::hex!["8eaf04151687736326c9fea17e25fc5287613693c912909cb226aa4794f26a48"]
+            .into();
+    let mut ext = TestExternalities::new_empty();
+    ext.register_extension(KeystoreExt(keystore.into()));
+    ext.execute_with(|| {
+        let submitter = ExecDelivery::select_authority(escrow.clone());
+
+        assert_eq!(submitter.is_ok(), true);
```
Comment (zannis) [inline circuit/src/execution-delivery/src/tests.rs:1306]: You can use the assert! macro when the value is expected to be true.

**Review by zannis — CHANGES_REQUESTED** [review]
Changes look ok, I would just add another unit test that covers the case where the submitter is not found and asserting the correct error.

**Review by zannis — APPROVED** [review]
Looking good!

File: circuit/src/execution-delivery/src/tests.rs (hunk 4 above)
Comment (beqaabu) [reply circuit/src/execution-delivery/src/tests.rs]: Added on the following commit

**Review by MaciejBaj — APPROVED** [review]

**Pull request merged**